core/bloombits/matcher.go

```diff
@@ -529,7 +529,8 @@ func (s *MatcherSession) Close() {
 	s.closer.Do(func() {
 		// Signal termination and wait for all goroutines to tear down
 		close(s.quit)
-		time.AfterFunc(time.Second, func() { close(s.kill) })
+		timeout := time.AfterFunc(time.Second, func() { close(s.kill) })
+		defer timeout.Stop()
```

Comment:
Something to note here: stopping the timer will prevent closing the `s.kill` channel if the WaitGroup unblocks before the timeout. This might be OK, but it means the `MatcherSession` can now be in two states after shutdown depending on the timeout.

Reply:
Is it ok waiting longer than timeout defined, if removed kill channel here.

Reply:
If you remove the `kill` channel and change all places where it waits on this channel, I think it will quit immediately. Then we don't need the timeout anymore.

Reply:
ok, i will make a change